src/app/modules/spcp/spcp.util.ts

```diff
@@ -33,7 +28,9 @@ import {
 } from './spcp.types'
 
 const logger = createLoggerWithLabel(module)
-const DESTINATION_REGEX = /^\/([\w]+)\/?/
+
+// Matches the MongoDB ObjectID hex format exactly (24 hex characters)
+const DESTINATION_REGEX = /^\/([a-fA-F0-9]{24})\/?$/
```

Comment:
`([a-fA-F0-9]{24})` is the form id pattern that is repeated in a bunch of places We could consider extracting it as a constant and use the Regexp constructor to compose the final pattern we want.
```javascript
const formIdPattern = ([a-fA-F0-9]{24});
const DESTINATION_REGEX = new RegExp(`/^\\/${formIdPattern}\\/?$/`);
```

Might not be worth it though. It's hard to read. String litteral regex patterns are always a pita with having to escape backslashes 😢